java/all/src/main/scala/org/typelevel/otel4s/java/OtelJava.scala

```diff
@@ -66,7 +66,9 @@ object OtelJava {
   def local[F[_]: Async: LocalContext](
       jOtel: JOpenTelemetry
   ): OtelJava[F] = {
-    val contextPropagators = new ContextPropagatorsImpl[F](jOtel.getPropagators)
+    val contextPropagators = ContextPropagators.of(
+      new TextMapPropagatorImpl(jOtel.getPropagators.getTextMapPropagator)
```

Comment:
it's worth noting that we don't currently expose a way to use `TextMapPropagator`s from our own API with the Java backend. it's orthogonal to this PR, but I thought it worth mentioning so others are aware

Reply:
>  don't currently expose a way to use

I slightly don't get it. Could you provide an example, please?

Reply:
suppose I write the following propagator:

```scala
import org.typelevel.otel4s.context.syntax._

final class FooBarPropagator[Ctx, K[X] <: Key[X]] private (
    ctxKey: K[Map[String, String]]
  )(implicit c: Contextual.Keyed[Ctx, K])
    extends TextMapPropagator[Ctx] {
  val fields = List("foo", "bar")

  def extract[A: TextMapGetter](ctx: Ctx, carrier: A): Ctx = {
    val map = fields
      .view
      .flatMap(k => TextMapGetter[A].get(carrier, k).map(v => k -> v))
      .toMap
    if (map.isEmpty) ctx else ctx.updated(ctxKey, map)
  }

  def inject[A: TextMapUpdater](ctx: Ctx, carrier: A): A =
    ctx.getOrElse(ctxKey, Map.empty)
      .iterator
      .foldLeft(carrier) { case (c, k -> v) => TextMapUpdater[A].updated(c, k, v) }
}

object FooBarPropagator {
  def apply[Ctx, K[X] <: Key[X]](implicit
    c: Contextual.Keyed[Ctx, K],
    kp: Key.Provider[SyncIO, K]
  ): FooBarPropagator[Ctx] =
    new FooBarPropagator(kp.uniqueKey[Map[String, String]]("foo-bar-propagation").unsafeRunSync())
}
```

there is currently no way (unless I've missed something) to use `FooBarPropagator` with `OtelJava`

Reply:
@NthPortal ah, I see. 

We can make it partially work if we provide a builder for `OtelJava`.

```scala
val otel4s: F[OtelJava[F]] = OtelJava
  .builder(useGlobal = true)
  .addPropagator(FooBarPropagator[...])
  .build
```

But extra propagators will only work with `Tracer[F].propagate` and `Tracer[F].joinOrRoot`.

Reply:
I'll see if I can write a reverse conversion from Scala to Java propagator. but that's for a separate PR

Reply:
Sounds good, thanks!